documentation/source/file.rst

```diff
@@ -1,8 +1,66 @@
 
 .. _file-example:
 
-doxygenfile Directive Example
-=============================
+doxygenfile Directive
+=====================
+
+This directive generates the appropriate output for a single source file. It
+takes the standard ``project``, ``path``, ``outline`` and ``no-link`` options
+and additionally the ``sections`` options.
+
+For the standard option refer to the documentation on the main directives page.
+The directive-specific options are documented below.
+
+``sections``
+   Limit the sections to render for the given source file to the given list.
+   Many of the names come from Doxygen and Breathe internals and may not make
+   sense from an external point of view. Nevertheless it is hoped this table is
+   useful.
+
+   .. csv-table:: Section types
+      :header: "Type", "Description"
+
+      "briefdescription", "Brief description"
+      "dcop-func", "DCOP Function"
+      "define", "Define"
+      "derivedcompoundref", "Derived compound reference"
+      "detaileddescription", "Detailed description"
+      "enum", "Enumerator"
+      "event", "Event"
+      "friend", "Friend"
+      "func", "Function"
+      "innerclass", "**Must be given to show sections inside a class**"
+      "innernamespace", "**Must be given to show sections inside a namespace**"
```

Comment:
Especially these 2, I assume this is the case based upon behaviour but perhaps you can verify?

Reply:
Thank you for going through the effort of making clean & thorough documentation!

Is there a button I need to press to approve this change?

Reply:
NVM, I see these changes in the files.